src/server/replica.cc

```diff
@@ -851,13 +851,47 @@ void Replica::RedisStreamAcksFb() {
   }
 }
 
+class AclCheckerClient : public ProtocolClient {
+ public:
+  AclCheckerClient(ServerContext server, Context* cntx)
+      : ProtocolClient(std::move(server)), cntx_(cntx) {
+    Connect();
+  }
+
+  void CheckAclRoundTrip() {
+    if (auto ec = SendCommandAndReadResponse(StrCat("REPLCONF acl-check ", "0")); ec) {
```

Comment:
Can you explain why we check "REPLCONF acl-check" ?
you want to check that the replica is not restricted to send REPLCONF command because REPLCONF ACK will fail and the master will not be updated with the lag and in this case you want to restart replication?
could a subcommand be restricted with acl i.e restrict only REPLCONF ACK ?

Reply:
> could a subcommand be restricted with acl i.e restrict only REPLCONF ACK ?

no it can't. The only subcommands you can currently restrict is ACLs and this is because each of the acl subcommands are stored in the command registry. This is not the case for other subcommands, which the main command is stored in the registry and when invoked act as a proxy to the rest of the subcommand familly. 


> will fail and the master will not be updated with the lag and in this case you want to restart replication 
Not really, you have two options:

1. Either reinstate the ACL credentials back like I do in the test example. Because the replica gets cancelled, the master node will try to reestablish a connection with the replica. It will fail every time until the ACL's for that user are reinstated. If they are reinstated, it will reconnect and continue replicating from where it was left off.
2. The replica operator decides to change the ACL user and reverify. In that case they need to call `config set` again to apply a new user and the `RELPICAOF` to reinstate the replication under a different user.

p.s. For (1) above it's the same with redis. Also in redis you need to restrict `psync` instead of `replconf` but this is natural because we have different execution paths and the internal commands are a little bit different.